runtime/src/qio/qio.c

```diff
@@ -2328,6 +2328,9 @@ qioerr _buffered_read_atleast(qio_channel_t* ch, int64_t amt)
     left -= num_read;
     qbuffer_iter_advance(&ch->buf, &read_start, num_read);
 
+    // Ignore interrupted system call, just keep reading.
+    if( err && qio_err_to_int(err) == EINTR ) err = 0;
+
```

Comment:
I'm not quite understanding the rationale behind this, but maybe I'm not picturing the situation correctly: my understanding is that an interrupt can occur when the user wishes to abort a process that is running.  In that situation, wouldn't we *want* the interrupt to stop us from continuing?